lib/ddtrace/propagation/http_propagator.rb

```diff
@@ -8,14 +8,10 @@ module HTTPPropagator
     include Ext::DistributedTracing
 
     # inject! popolates the env with span ID, trace ID and sampling priority
-    def self.inject!(span, env)
-      headers = { HTTP_HEADER_TRACE_ID => span.trace_id.to_s,
-                  HTTP_HEADER_PARENT_ID => span.span_id.to_s }
-      if span.sampling_priority
-        headers[HTTP_HEADER_SAMPLING_PRIORITY] = span.sampling_priority.to_s
-      end
-      env.merge! headers
-      env.delete(HTTP_HEADER_SAMPLING_PRIORITY) unless span.sampling_priority
+    def self.inject!(span, context, env)
```

Comment:
So I would not add he context as an arg to this function. Because, we're going to remove it some day, for sure. The final state is something like: context has it all, trace_id, current span_id, and sampling priority. As in https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/context.py#L25

So in the current state, the Ruby impl. does not have full context support. Initially, I thought it was OK to keep the legacy "span-based" way of doing things. Which, actually, works, only you copy the information many times in all spans when once in the context should be enough. My point is: we need to choose, either keep it all in span, or move it all in context, but something in between is really uncomfortable. AFAIK in different places in Python/Ruby code there's this "it could be a span or a context, let's test at run-time" as in https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/tracer.py#L159 and I think this is really fine. But two args (span+context) will be troublesome to migrate I fear.

Reply:
We should keep the compatibility that simplify our migration to OpenTracing (while not breaking a lot of stuff). In that case, the public API expects a `span_context` that in our case is our `Context` (this is an approximation like @ufoot said). Eventually the `Span` will have its `SpanContext` and our (Trace)`Context` will only be an implementation detail. For instance, In Python we're providing the propagator in that way: https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/propagation/http.py

Reference for OpenTracing: https://github.com/opentracing/basictracer-python/blob/master/basictracer/tracer.py#L87

I'd suggest to expect a `Context` (so `span.context` that will be easy to migrate later) so:
* the API is compliant with our Python implementation + OpenTracing
* no confusion so users always pass a `Context`
* since it's a new functionality, no breaking changes in the future

Reply:
@ufoot @palazzem OK, makes total sense. I will change things so we receive only the context than.